src/app/ClusterStatus/ClusterStatus.tsx

```diff
@@ -21,108 +21,65 @@ import {
   Title
 } from '@patternfly/react-core';
 import { CubesIcon, SearchIcon } from '@patternfly/react-icons';
-import { Table, TableBody, TableHeader, TableVariant } from '@patternfly/react-table';
+import { TableComposable, Thead, Tr, Th, Tbody, Td, IAction, ActionsColumn } from '@patternfly/react-table';
 import { Health } from '@app/Common/Health';
 import { useApiAlert } from '@app/utils/useApiAlert';
 import { TableErrorState } from '@app/Common/TableErrorState';
 import { useTranslation } from 'react-i18next';
 import { ConsoleServices } from '@services/ConsoleServices';
+import { useFetchClusterMembers } from '@app/services/clusterHook';
 
 const ClusterStatus: React.FunctionComponent<any> = (props) => {
   const { addAlert } = useApiAlert();
   const { t } = useTranslation();
   const brandname = t('brandname.brandname');
-  const [error, setError] = useState<undefined | string>();
-  const [loading, setLoading] = useState<boolean>(true);
-  const [cacheManager, setCacheManager] = useState<undefined | CacheManager>(undefined);
+  const { clusterMembers, cacheManager, loading, error, reload } = useFetchClusterMembers();
   const [filteredClusterMembers, setFilteredClusterMembers] = useState<ClusterMember[]>([]);
   const [clusterMembersPagination, setClusterMembersPagination] = useState({
     page: 1,
     perPage: 10
   });
-  const [rows, setRows] = useState<(string | any)[]>([]);
-  const columns = [
-    { title: 'Name' },
+
+  const columnNames = {
+    name: t('cluster-membership.node-name'),
+    physicalAdd: t('cluster-membership.physical-address')
+  };
+
+  const clusterMemberActions = (row): IAction[] => [
     {
-      title: 'Physical address'
+      title: t('cluster-membership.download-report'),
+      onClick: () => {
+        ConsoleServices.server().downloadReport(row.name);
```

Comment:
can;t this be done with a hook ?
would be nice to try to unit test with jest mocking the server report download calls

Reply:
Added hook and unit test.